src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -2390,6 +2398,7 @@ static void close_from_api(grpc_chttp2_transport* t, grpc_chttp2_stream* s,
   grpc_slice_buffer_add(&t->qbuf, status_hdr);
   grpc_slice_buffer_add(&t->qbuf, message_pfx);
   grpc_slice_buffer_add(&t->qbuf, grpc_slice_ref_internal(slice));
+  grpc_chttp2_reset_ping_clock(t);
```

Comment:
As I understand the problem here was that the server thought that the client was violating GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS ?

If so I'm a little confused by the documentation around `GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS`, that are described in https://github.com/grpc/grpc/blob/master/doc/keepalive.md.

Mainly this wording: "If there are no data frames being sent on the transport", makes me think that the transport only cares <i>data frames</i>. Should the wording there be expanded beyond data frames to all frame types?

Reply:
You are right. (Minor correction - "all frame types" - "data/header/window_update"). Created #23151